docs/ml-features.md

```diff
@@ -217,21 +217,27 @@ for feature in result.select("result").take(3):
 
 [Tokenization](http://en.wikipedia.org/wiki/Lexical_analysis#Tokenization) is the process of taking text (such as a sentence) and breaking it into individual terms (usually words).  A simple [Tokenizer](api/scala/index.html#org.apache.spark.ml.feature.Tokenizer) class provides this functionality.  The example below shows how to split sentences into sequences of words.
 
-Note: A more advanced tokenizer is provided via [RegexTokenizer](api/scala/index.html#org.apache.spark.ml.feature.RegexTokenizer).
+[RegexTokenizer](api/scala/index.html#org.apache.spark.ml.feature.RegexTokenizer) allows more advanced tokenization based on regular expression (regex) matching. By default, the parameter pattern (regex, default: \\s+) is used as delimiters to split the input text. Alternatively, users can set parameter gaps to false indicating the regex pattern denotes "tokens" rather than splitting "gaps", and find all matching occurrences as the tokenization result.
```

Comment:
minor: try to limit the line width to 100 and make new line at the end of each sentence. This helps reduce the diff size when new changes come in.

Quote `pattern` and `gaps` so users know that they are parameter names.